controller/konnect/reconciler_consumerref.go

```diff
@@ -103,17 +103,6 @@ func handleKongConsumerRef[T constraints.SupportedKonnectEntityType, TEnt constr
 		return ctrl.Result{Requeue: true}, nil
 	}
 
-	old := ent.DeepCopyObject().(TEnt)
-	if err := controllerutil.SetOwnerReference(&consumer, ent, cl.Scheme(), controllerutil.WithBlockOwnerDeletion(true)); err != nil {
```

Comment:
Have we considered what our behavior should be if there are already existing credentials under consumer ownership (assigned by KGO 1.4)? I understand that in the current approach, we accept that objects reconciled by KGO 1.4 will end with cascade delete if a consumer gets deleted, right?

WDYT about an alternative where we migrate the already reconciled objects by removing their owner references?

Reply:
That's a good point. I'll address that. I think that simply calling `controllerutil.RemoveOwnerReference()` would do. WDYT?